hypothesis-python/src/hypothesis/extra/numpy.py

```diff
@@ -649,3 +649,52 @@ def nested_dtypes(
     return st.recursive(
         subtype_strategy, lambda x: array_dtypes(x, allow_subarrays=True), max_leaves
     ).filter(lambda d: max_itemsize is None or d.itemsize <= max_itemsize)
+
+
+@st.defines_strategy
+def valid_tuple_axes(ndim, min_size=0, max_size=None):
+    # type: (int, int, int) -> st.SearchStrategy[Tuple[int, ...]]
+    """Return a strategy for generating permissible tuple-values for the
+    ``axis`` argument for a numpy sequential function (e.g.
+    :func: `numpy:numpy.sum`), given an array of the specified
+    dimensionality.
+
+    All tuples will have an length >= min_size and <= max_size. The default
+    value for max_len is ``ndim``.
+
+    Examples from this strategy shrink towards an empty tuple, which render
+    most sequential functions as no-ops.
+
+    The following are some examples drawn from this strategy.
+
+    .. code-block:: pycon
+
+        >>> [valid_tuple_axes(3).example() for i in range(4)]
+        [(-3, 1), (0, 1, -1), (0, 2), (0, -2, 2)]
+
+    ``valid_tuple_axes`` can be joined with other strategies to generate
+    any type of valid axis object, i.e. integers, tuples, and ``None``:
+
+    .. code-block:: pycon
+
+        >>> from hypothesis.strategies import integers, none
+        >>> from hypothesis.extra.numpy import valid_tuple_axes
+        >>> def valid_axes(ndim):
+        ...     return none() | integers(-ndim, ndim - 1) | valid_tuple_axes(ndim)
```

Comment:
We *mostly* don't bother showing imports in small examples - in this case the `st.` prefix is enough, and readers should know where `valid_tuple_axes` is from!  e.g.

```rst
.. code-block:: python
    any_axis_strategy = none() | integers(-ndim, ndim - 1) | valid_tuple_axes(ndim)
```